pkg/runtime/serializer/streaming/streaming.go

```diff
@@ -83,9 +83,9 @@ func (d *decoder) Decode(defaults *unversioned.GroupVersionKind, into runtime.Ob
 				continue
 			}
 			// double the buffer size up to maxBytes
-			if cap(d.buf) < d.maxBytes {
+			if len(d.buf) < d.maxBytes {
```

Comment:
And is there a guarantee that it will always be "full" if ErrShortBuffer is returned?